File: PO_Projekt3/wilk.py
from zwierze import Zwierze

class Wilk(Zwierze):
    def __init__(self, swiat, polozenie):
        super().__init__(swiat, polozenie, 9, 5, 1.00, 1, 0.00, 1.00)

    def tegoSamegoTypu(self, inny):
        return isinstance(inny, Wilk)

    def stworzDziecko(self, cel):
        return Wilk(self._swiat, cel)

    def rysowanie(self):
        return (204, 204, 204)

File: projekty/PO_Projekt2/ProgOb_Projekt2/src/main/java/pl/Dybowski/Adrian/Projekt2/WirtualnySwiat/Zwierzeta/Wilk.java
package pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Zwierzeta;

import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Organizm;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Stale;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Swiat;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Wspolrzedne2D;

public class Wilk extends Zwierze {	
	public Wilk(Swiat swiat, Wspolrzedne2D polozenie) {
		super(swiat, polozenie, Stale.sila.get(Wilk.class), Stale.inicjatywa.get(Wilk.class),
				Stale.szansaRuchu.get(Wilk.class), Stale.zasiegRuchu.get(Wilk.class),
				Stale.szansaUcieczki.get(Wilk.class), Stale.szansaRozmnazania.get(Wilk.class));
	}

	@Override
	public boolean tegoSamegoTypu(Organizm inny) {
		return inny instanceof Wilk;
	}

	@Override
	public Organizm stworzDziecko(Wspolrzedne2D cel) {
		return new Wilk(swiat, cel);
	}
}
